src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/Completion.cs

```diff
@@ -122,16 +124,27 @@ internal partial class RazorCustomMessageTarget
 
         try
         {
+            using var _1 = ArrayBuilderPool<CompletionItem>.GetPooledObject(out var builder);
```

Comment:
You should be able to use a `PooledArrayBuilder`. That was recently changed to a normal struct to allow it to be usable in async methods.